python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -573,6 +573,7 @@ class RandomForestClassifier(Base, ClassifierMixin):
             memory used for the method.
         """
         self._set_output_type(X)
+        self._set_target_dtype(y)
         self._set_n_features_in(X)
```

Comment:
I wonder if it would be worth creating a function that encompasses this 3 calls, something like 

```python
self._set_base_attributes(X, output_type=True, dtype=True, n_features=True)
```
kinda thing so that instead of three calls, estimators have a single call, and we can add more attributes without increasing the number of lines inheritting attributes need to call

Reply:
I think this makes a lot of sense. It's in the same spirit of how sklearn wraps a lot of logic into `_validate_data` on the base class

Reply:
Would you prefer that in a follow-up or to change the scope of this PR?

Reply:
Going to file an issue for a follow-up

Reply:
Filed as #2488